Assets/Scripts/Jump.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Jump : MonoBehaviour {
+	
+	public float jumpSpeed = 8.0f;
+
+	private Vector3 moveDirection = Vector3.zero;
+	private bool grounded = true;
+	private InputController inputController;
+	private int playerNum;
+	private string jumpAxis;
+	// Use this for initialization
+	void Start () {
+		inputController = GameObject.FindWithTag ("GameController").GetComponent<InputController>();
+		string parentTag = this.transform.parent.tag;
+		string lastChar = parentTag.Substring (parentTag.Length - 1);
+		int.TryParse (lastChar, out playerNum);
+		jumpAxis = inputController.GetThrowInput (playerNum);
+	}
+	
+	void FixedUpdate() 
+	{
+		Jumping();
+	}
+
+	void Jumping() {
+		if (grounded)
+		{
+			float joystickThrow = Input.GetAxis (jumpAxis);
```

Comment:
`joystickThrow` is a bit perplexing here. Should this be `jumpDirection` or `jumpVector`? This may be less confusing if we rename all instances of "throw" to "fling" to more appropriately describe our in-game verbs.